### PR TITLE
feat(RELEASE-230): update release service ref to konflux ci org

### PR DESCRIPTION
**Pull request by happybhati**
This commit updates the `release-service` reference
 as result of migration to `konflux-ci` from `redhat-appstudio`
 more details parent EPIC:
 https://issues.redhat.com/browse/RHTAPREL-800

### CODE REVIEW

**Review by johnbieren — COMMENTED** [review]
We also need the CODEOWNERS updated

File: .github/CODEOWNERS
Comment (johnbieren) [inline .github/CODEOWNERS:1]: This is throwing an error

File: go.sum
Comment (johnbieren) [inline go.sum:303]: This doesn't look right

File: .github/CODEOWNERS
Comment (happybhati) [reply .github/CODEOWNERS:1]: It should be fixed now; updated team permissions!
Thank you!

File: go.sum
Comment (happybhati) [reply go.sum:303]: When I ran `go mod tidy,` the following lines were added.
```
github.com/redhat-appstudio/release-service v0.0.0-20230511145849-bde1cdcbb60b h1:96jgqIR8Otx2vrLZscvNo3gFlX2d9P/mkbx5Trkm8tQ=
```

I am not sure when the new package will be available for konflux-ci/release-service
like this https://pkg.go.dev/github.com/redhat-appstudio/release-service/api/v1alpha1 

Once we merge this PR? with the updated module name?

Comment (johnbieren) [reply go.sum:303]: Okay, can you put investigating once this PR merges if we can remove this line and if not, say why on your todo list?

Comment (happybhati) [reply go.sum:303]: Definitely!

**Review by johnbieren — APPROVED** [review]

**Comment by happybhati** [comment]
I am merging this PR. All checks pass except `ci/prow/release-service-e2e`, possibly because of the interdependence on this change.

**Pull request merged**
